### PR TITLE
minimal installer: support cloud-init

### LINKED ISSUE #39076: Add cloud-init service to minimal ISO
There are other cloud providers than AWS that support the userdata system.

Since the minimal ISO works everywhere, I propose that the ISO gets `services.cloud-init.enabled = true;`. If there is no config source, nothing will happen, and thus the minimal ISO can be used to bootstrap on cloud services.

### PR DESCRIPTION
**Pull request by wmertens**
Fixes #39076 

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: nixos/modules/installer/cd-dvd/installation-cd-minimal.nix [hunk 1]
```diff
@@ -7,4 +7,7 @@
   imports =
     [ ./installation-cd-base.nix
     ];
+
+  # Enable cloud-init so the image can also be used for cloud providers
+  services.cloud-init.enable = true;
```
Comment (Mic92) [inline nixos/modules/installer/cd-dvd/installation-cd-minimal.nix]: This adds python + libraries to the iso. How much does this increase the ISO?

**Comment by Mic92** [comment]
We have `modules/virtualisation/nova-config.nix` do we offer a `cloud image` somewhere?

File: nixos/modules/installer/cd-dvd/installation-cd-minimal.nix (hunk 1 above)
Comment (wmertens) [reply nixos/modules/installer/cd-dvd/installation-cd-minimal.nix]: Python 2.7 is already in the minimal image, so I'm hoping not much. I hope I can find out via ofBorg :)

Comment (wmertens) [reply nixos/modules/installer/cd-dvd/installation-cd-minimal.nix]: ouch, running `nix-shell -p cloud-init` on the minimal ISO turns into a 60MB download and 500MB unpacked :-/

Not sure if that is the best way to test though.

**Comment by wmertens** [comment]
Another approach would be to have a minimal script like the one @copumpkin wrote for AWS. Parse at least the ssh stuff and expect a nix expression for the rest. User data is free-form…

**Comment by wmertens** [comment]
@GrahamcOfBorg build nixos.iso_minimal.x86_64-linux

**Comment by wmertens** [comment]
@GrahamcOfBorg build iso_minimal.x86_64-linux

**Comment by dezgeg** [comment]
Doesn't cloud-init block bootup until it manages to reach the metadata server? I can't recall exactly where I encountered that but IIRC it does.

**Comment by wmertens** [comment]
I'm closing this, I'll continue the conversation in #39076 to come up with something better suited to NixOS.

**Pull request closed without merging**